aicsimageio/readers/glob_reader.py

```diff
@@ -0,0 +1,541 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from typing import Any, Dict, List, Optional, Tuple, Union, Callable
+
+import re
+import glob
+from collections import OrderedDict
+from pathlib import Path
+import dask.array as da
+import numpy as np
+import pandas as pd
+import xarray as xr
+from dask import delayed
+from fsspec.spec import AbstractFileSystem
+from tifffile import TiffFile, TiffFileError, imread
+from tifffile.tifffile import TiffTags
+
+from .. import constants, exceptions, types
+from ..dimensions import (
+    DEFAULT_CHUNK_DIMS,
+    REQUIRED_CHUNK_DIMS,
+    DimensionNames,
+    DEFAULT_DIMENSION_ORDER,
+    DEFAULT_DIMENSION_ORDER_LIST_WITH_SAMPLES,
+)
+from ..metadata import utils as metadata_utils
+from ..utils import io_utils
+from .reader import Reader
+
+
+TIFF_IMAGE_DESCRIPTION_TAG_INDEX = 270
+
+
+class GlobReader(Reader):
```

Comment:
I would suggest we call this GlobTiffReader or TiffGlobReader since it is really not built to handle anything else.  Also, I feel the code that checks for whether to use this reader should also check to make sure ".tif/.tiff" appears in the paths.  To really drive home that this won't work for anything else.

Reply:
Agree on renaming this to `TiffGlobReader`, disagree on checking the path because its pretty common in object storage to not have a suffix on the URI. Plus we are already checking the bytes :shrug: